gnowledge/gstudio
# Implemented filtering module in E-Library

**Implemented filtering in E-Library:** 
- For time being filters are applied to e-library app only because:
  - filters in `File` app are different.
  - In future only one function will handle this in the AJAX way. 
- UI for filters improved to facilitate filtering smoothly.
- ndf_tag is added to get the filters data.
- The redundant code in e-library is reduced to great amount.

--

**Future expected improvements:**
- Filters for File app.
- In future file app and e-library app needs to be unified with single method with proper arguments to query.
- Filters will load from AJAX.
- Listing of fields along with attributes in filters category.
- If possible text search in filter selection dropdown.
- Reset filters button.

--

**TEST:**
- Go to e-library and try different filters.
- Remove filters and apply filters.
- Check for change in counts of `educationaluse` categories listed on LHS.
